### DISCUSSION TITLE
Preventing `addCollection` from generating a tag?

### DISCUSSION THREAD

**Discussion by j9t** [11ty/eleventy#2792] [Q&A]
In .eleventy.js, I’m using `addCollection` like the following:

```js
eleventyConfig.addCollection("postsByPubDate", function(collectionApi) {
  return collectionApi.getFilteredByTag("posts").sort((a, b) => (Number(a.data.date_publication) > Number(b.data.date_publication) ? 1 : -1));
});
```

What’s unexpected and strange is that this seems to create a tag, “postsByPubDate”. Is this a bug—or, if not, how can this be suppressed not to need to manually exclude this and similar collections from `filterTagList`?

**Comment by danburzo** [comment]
Are you building on top of [11ty/eleventy-base-blog](https://github.com/11ty/eleventy-base-blog), which seems to be using a `filterTagList` filter? When you say _this seems to create a tag_, which piece of template code prints `postsByPubDate` unexpectedly?

Or, if you mean that you see `tags/postsByPubDate/index.html` in the output, that is because the thing that generates individual tags, [`content/tags.njk`](https://github.com/11ty/eleventy-base-blog/blob/main/content/tags.njk), does so by paginating `collections`, which includes `postsByPubDate`, and which you'll need to filter out.

**Reply by j9t** [reply]
Yes, this had been building on eleventy-base-blog, and the behavior is as you described, there’s an unexpected tags/postsByPubDate/index.html.

Thank you for the swift reply—it seems this is to be expected, but I’ll try the alternative you sent.

**Comment by danburzo** [comment]
If you want to create individual tag pages that reflect strictly the set of `tags` that are present in your content's `data` without other complications, you might want to create a `tags` collection:

```js
eleventyConfig.addCollection('tags', api => {
	return Array.from(
		new Set(
			api.getAll()
				.map(it => it.data.tags ?? [])
				.flat()
		)
	);
});
```

Then change `tags.njk` to paginate the `tags` collection instead of the set of all collections.

```diff
 ---
 pagination:
-  data: collections
+  data: collections.tags
   size: 1
   alias: tag
-  filter:
-    - all
-    - post
-    - posts
-    - tagList
   addAllPagesToCollections: true
 layout: layouts/home.njk
 eleventyComputed:
```

**Comment by pdehaan** [comment]
Yeah, I don't think I'm seeing the described behavior. I created a simple 3 page site w/ a `posts` tag in a directory data file and then used your `postsByPubDate` custom collection code but didn't notice it adding new tags to pages.

```js
eleventyConfig.addCollection("postsByPubDate", function(collectionApi) {
  return collectionApi.getFilteredByTag("posts").sort((a, b) => (Number(a.data.date_publication) > Number(b.data.date_publication) ? 1 : -1));
});
```

**Reply by j9t** [reply]
Maybe this is connected to the fact that this had been based on eleventy-base-blog. As per @danburzo, this seems like it’s to be expected, and I’ll try the suggested change.

(I wonder if there’s something to improve in the template, as that coupling between https://github.com/11ty/eleventy-base-blog/blob/main/eleventy.config.js#L75 and https://github.com/11ty/eleventy-base-blog/blob/main/content/tags.njk#L7 makes for a trap, too, and isn’t maintainable.)

**Reply by danburzo** [reply]
I believe that way of implementing tag pages is in support of this article: [Quick Tip #004—Zero Maintenance Tag Pages for your Blog](https://www.11ty.dev/docs/quicktips/tag-pages/). That is, a way to generate tag pages without ever having to touch the Eleventy config. However, the solution does require, in practice, you to be vigilant about filtering out any new collection you add.
